### PR TITLE
Fix numerics for activation recompute

### PR DESCRIPTION
**Pull request by ksivaman**
This PR is a numerical fix to ensure that the results are bitwise identical when running TE (w/o FP8) with and without activation recompute.

- Add missing warmup for kernel with no bias.
- Fix incorrect input shape during jit warmup.

### CODE REVIEW

File: transformer_engine/pytorch/jit.py [hunk 1]
```diff
@@ -229,8 +229,9 @@ def warmup_jit_bias_gelu(
     for bias_grad, input_grad in zip([True, True], [False, True]):
         bias.requires_grad, inp.requires_grad = bias_grad, input_grad
         for _ in range(5):
-            output = bias_gelu_fused(inp, bias)
-    del bias, inp, output
+            _ = bias_gelu_fused_(inp, bias)
+            _ = gelu_fused_(inp)
```
Comment (timmoon10) [inline transformer_engine/pytorch/jit.py:233]:
```suggestion
            bias_gelu_fused_(inp, bias)
            gelu_fused_(inp)
```

**Review by timmoon10 — APPROVED** [review]
LGTM
